lightly/api/api_workflow_predictions.py

```diff
@@ -0,0 +1,155 @@
+from concurrent.futures import ThreadPoolExecutor
+from typing import Dict, List, Union, Optional
+
+import tqdm
+
+from lightly.api.prediction_singletons import PredictionSingletonClassificationRepr
+from lightly.api.utils import retry
+from lightly.openapi_generated.swagger_client import (
+    PredictionTaskSchema,
+    SamplePartialMode,
+    PredictionSingletonClassification,
+    PredictionSingletonObjectDetection,
+    PredictionSingletonInstanceSegmentation,
+    PredictionSingletonKeypointDetection,
+)
+
+
+class _PredictionsMixin:
+    def create_or_update_prediction_task_schema(
+        self,
+        schema: PredictionTaskSchema,
+        prediction_uuid_timestamp: int,
+    ) -> None:
+        """Creates or updates the prediction task schema
+
+        Args:
+            schema:
+                The prediction task schema.
+            prediction_uuid_timestamp:
+                The uuid timestamp in nanoseconds since 1970. Get it via 'int(time.time())'.
+
+        Example:
+          >>> import time
+          >>> from lightly.api import ApiWorkflowClient
+          >>> from lightly.openapi_generated.swagger_client import (
+          >>>     PredictionTaskSchema,
+          >>>     TaskType,
+          >>>     PredictionTaskSchemaCategory,
+          >>> )
+          >>>
+          >>> client = ApiWorkflowClient(
+          >>>     token="MY_LIGHTLY_TOKEN", dataset_id="MY_DATASET_ID"
+          >>> )
+          >>>
+          >>> timestamp = int(time.time())
+          >>> schema = PredictionTaskSchema(
+          >>>     name="my-object-detection",
+          >>>     type=TaskType.OBJECT_DETECTION,
+          >>>     categories=[
+          >>>         PredictionTaskSchemaCategory(id=0, name="dog"),
+          >>>         PredictionTaskSchemaCategory(id=1, name="cat"),
+          >>>     ],
+          >>> )
+          >>> client.create_or_update_prediction_task_schema(
+          >>>     schema=schema, prediction_uuid_timestamp=timestamp
+          >>> )
+
+
+        """
+        self._predictions_api.create_or_update_prediction_task_schema_by_dataset_id(
+            body=schema,
+            dataset_id=self.dataset_id,
+            prediction_uuid_timestamp=prediction_uuid_timestamp,
+        )
+
+
+    def create_or_update_predictions(
+        self,
+        filename_to_prediction_singletons: Union[
```

Comment:
AFAIK this should be a `Dict[str, List[Union[PredictionSingletonClassificationRepr, PredictionSingletonObjectDetection, PredictionSingletonInstanceSegmentation, PredictionSingletonKeypointDetection]]`, no? We have singletons for each file and these singletons can be mixed if multiple tasks are done, no?

Reply:
Yes, I changed that.